pos_membership_delegated_partner/static/src/xml/Orderline.xml

```diff
@@ -0,0 +1,42 @@
+<?xml version="1.0" encoding="UTF-8" ?>
+<templates id="template" xml:space="preserve">
+
+    <t
+        t-name="Orderline"
+        t-inherit="point_of_sale.Orderline"
+        t-inherit-mode="extension"
+        owl="1"
+    >
+        <xpath expr="//ul[hasclass('info-list')]" position="inside">
+            <li class="info orderline-membership" t-if="props.line.product.membership">
+                <t t-if="props.line.delegated_member">
+                    <i
+                        class="fa fa-hand-o-right"
+                        role="img"
+                        aria-label="Member"
+                        title="Member"
+                    />
+                    <t t-out="props.line.delegated_member.name" />
+                </t>
+                <t t-else="">
+                    <t t-if="props.line.order.get_partner()">
+                        <i
+                            class="fa fa-hand-o-right"
+                            role="img"
+                            aria-label="Member"
+                            title="Member"
+                        />
+                        <t t-out="props.line.order.get_partner().name" />
+                        <span> (<i
+                                class="fa fa-user"
+                                role="img"
+                                aria-label="Customer"
+                                title="Customer"
+                            />)</span>
+                    </t>
+                </t>
```

Comment:
instead of having a big `t-if`/`t-else` with almost the same thing in both parts, the orderline should have a method returning the membership partner (with the logic there). the display of the customer icon could be in an `t-if="!props.line.delegated_member"`.